packages/jsii-pacmak/lib/generator.ts

```diff
@@ -48,18 +48,54 @@ export interface IGenerator {
   save(outdir: string, tarball: string): Promise<any>;
 }
 
+export abstract class GeneratorBase implements IGenerator {
+  abstract generate(fingerprint: boolean): void
+  abstract async load(packageDir: string, assembly: reflect.Assembly): Promise<void>
+  abstract async upToDate(outDir: string): Promise<boolean>
+  abstract async save(outdir: string, tarball: string): Promise<any>
```

Comment:
Make sure to have visibility modifiers consistently (I reckon this are `public`).